src/components/RangeInput/RangeInput.tsx

```diff
@@ -1,49 +1,62 @@
 /** @jsx h */
 
 import { h, Component } from 'preact';
-import PropTypes from 'prop-types';
 import cx from 'classnames';
 import Template from '../Template/Template';
+import {
+  RangeInputCSSClasses,
+  RangeInputTemplates,
+} from '../../widgets/range-input/range-input';
+import { Range, RangeBoundaries } from '../../connectors/range/connectRange';
+
+export type RangeInputComponentCSSClasses = {
+  [TClassName in keyof RangeInputCSSClasses]: string;
+};
 
-class RangeInput extends Component {
-  constructor(props) {
-    super(props);
+export type RangeInputProps = {
+  min?: number;
+  max?: number;
+  step: number;
+  values: Partial<Range>;
+  cssClasses: RangeInputComponentCSSClasses;
+  templateProps: {
+    templates: RangeInputTemplates;
+  };
+  refine(rangeValue: RangeBoundaries): void;
+};
 
-    this.state = {
-      min: props.values.min,
-      max: props.values.max,
-    };
-  }
+class RangeInput extends Component<RangeInputProps, Partial<Range>> {
+  public state = {
+    min: this.props.values.min,
+    max: this.props.values.max,
+  };
 
-  componentWillReceiveProps(nextProps) {
+  public componentWillReceiveProps(nextProps: RangeInputProps) {
     this.setState({
       min: nextProps.values.min,
       max: nextProps.values.max,
     });
   }
 
-  onInput = name => event => {
+  private onInput = (key: string) => (event: Event) => {
+    const { value } = event.currentTarget as HTMLInputElement;
+
     this.setState({
-      [name]: event.currentTarget.value,
+      [key]: Number(value),
     });
   };
 
-  onSubmit = event => {
+  private onSubmit = (event: Event) => {
     event.preventDefault();
 
-    this.props.refine([
-      this.state.min && Number(this.state.min),
-      this.state.max && Number(this.state.max),
-    ]);
+    this.props.refine([this.state.min, this.state.max]);
   };
 
-  render() {
+  public render() {
     const { min: minValue, max: maxValue } = this.state;
     const { min, max, step, cssClasses, templateProps } = this.props;
-    const isDisabled = min >= max;
-
+    const isDisabled = min && max ? min >= max : false;
```

Comment:
I checked, and this behaves the same, undefined on either side of a comparison will cause false, so this is correct!